supreme/src/iosMain/kotlin/at/asitplus/signum/supreme/os/IosKeychainProvider.kt

```diff
@@ -0,0 +1,628 @@
+                throw NoSuchElementException("Key with alias $alias already exists")
+        }
+        deleteSigningKey(alias).getOrThrow() /* make sure there are no leftover private keys */
+
+        val config = DSL.resolve(::IosSigningKeyConfiguration, configure)
+
+        val availability = config.hardware.v.let { c-> when (c.availability) {
+            IosSecureEnclaveConfiguration.Availability.ALWAYS -> if (c.allowBackup) kSecAttrAccessibleAlways else kSecAttrAccessibleAlwaysThisDeviceOnly
+            IosSecureEnclaveConfiguration.Availability.AFTER_FIRST_UNLOCK -> if (c.allowBackup) kSecAttrAccessibleAfterFirstUnlock else kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+            IosSecureEnclaveConfiguration.Availability.WHILE_UNLOCKED -> if (c.allowBackup) kSecAttrAccessibleWhenUnlocked else kSecAttrAccessibleWhenUnlockedThisDeviceOnly
+        } }
+
+        val useSecureEnclave = when (config.hardware.v.backing) {
+            is REQUIRED -> true
+            is PREFERRED -> isSecureEnclaveSupportedConfiguration(config._algSpecific.v)
+            is DISCOURAGED -> false
+        }
+
+        val publicKeyBytes: ByteArray = memScoped {
+            val attr = createCFDictionary {
+                when (val alg = config._algSpecific.v) {
+                    is SigningKeyConfiguration.ECConfiguration -> {
+                        kSecAttrKeyType mapsTo kSecAttrKeyTypeEC
+                        kSecAttrKeySizeInBits mapsTo alg.curve.coordinateLength.bits.toInt()
+                    }
+                    is SigningKeyConfiguration.RSAConfiguration -> {
+                        kSecAttrKeyType mapsTo kSecAttrKeyTypeRSA
+                        kSecAttrKeySizeInBits mapsTo alg.bits
+                    }
+                }
+                if (useSecureEnclave) {
+                    kSecAttrTokenID mapsTo kSecAttrTokenIDSecureEnclave
+                }
+                kSecPrivateKeyAttrs mapsTo createCFDictionary {
+                    kSecAttrApplicationLabel mapsTo alias
+                    kSecAttrIsPermanent mapsTo true
+                    kSecAttrApplicationTag mapsTo KeychainTags.PRIVATE_KEYS
+                    when (val hwProtection = config.hardware.v.protection.v) {
+                        null -> {
+                            kSecAttrAccessible mapsTo availability
+                        }
+                        else -> {
+                            val factors = hwProtection.factors.v
+                            kSecAttrAccessControl mapsTo corecall {
+                                SecAccessControlCreateWithFlags(
+                                    null, availability,
+                                    when {
+                                        (factors.biometry && factors.deviceLock) -> kSecAccessControlUserPresence
+                                        factors.biometry -> kSecAccessControlBiometryAny
+                                        else -> kSecAccessControlDevicePasscode
```

Comment:
I am still not sure about this. on iOS you can chose to require either _any_ or _all_ specified auth methods. But this is probably a case for an issue and not for this PR

Reply:
see #108